zipline/utils/events.py

```diff
@@ -539,7 +539,7 @@ def get_first_trading_day_of_month(self, dt):
         self.month = dt.month
 
         dt = dt.replace(day=1)
-        first_day = (dt if self.cal.is_open_on_day(dt)
+        first_day = (normalize_date(dt) if self.cal.is_open_on_day(dt)
```

Comment:
OK, I'm fine leaving this as-is as long as we're aware that it's broken for nonstandard calendars and we're working on a fix.